internal/resolution/variablesources/bundle_deployment.go

```diff
@@ -0,0 +1,58 @@
+package variablesources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ input.VariableSource = &BundleDeploymentVariableSource{}
+
+type BundleDeploymentVariableSource struct {
+	client              client.Client
+	inputVariableSource input.VariableSource
+}
+
+func NewBundleDeploymentVariableSource(cl client.Client, inputVariableSource input.VariableSource) *BundleDeploymentVariableSource {
+	return &BundleDeploymentVariableSource{
+		client:              cl,
+		inputVariableSource: inputVariableSource,
+	}
+}
+
+func (o *BundleDeploymentVariableSource) GetVariables(ctx context.Context, entitySource input.EntitySource) ([]deppy.Variable, error) {
```

Comment:
we should probably have good coverage of this function, wdyt?

Reply:
done, CI failing due to codecov issues..